Sources/SwiftCrossUI/Views/NavigationStack.swift

```diff
@@ -0,0 +1,188 @@
+import Gtk
+
+/// Type to indicate the root of the NavigationStack. This is private to prevent root accidentally showing instead of a detail view.
+private struct NavigationStackRootPath: Hashable {}
```

Comment:
Technically probably yes but this would deviate the implementation quite far from ForEach and I spent some time figuring out how to make that work but I can't crack it without making the logic hard to maintain.